confmap/provider/eecprovider/README.md

```diff
@@ -0,0 +1,28 @@
+# EEC Provider
+
+This is an OpenTelemetry Collector `confmap.Provider` module that allows the
+Collector to be configured with the Dynatrace Extensions Execution Controller.
+
+> [!WARNING]
+> This is not intended for customer use, but is only intended for use by
+> Dynatrace when managing its own Collector deployments. As such, issues with
+> this component will fall outside our support policy.
+
+## Configuration
+
+This confmap Provider can be configured by pasing query parameter-formatted values
+inside the fragment of the URL given to the config flag. For example:
+
+```text
+dynatrace-otel-collector --config=eec://my.eec.host:31098#refresh-interval=5s&auth-header=Api-Token&auth-file=/var/private/token.key
+```
+
+### Options
+
+| Key | Default | Description |
+|-----|---------|-------------|
+| auth-env |  None | An environment variable that should be read to get a plaintext API token or other key to be used in an HTTP header. Mutually exclusive from `auth-file`, passing both options should result in an error. |
+| auth-file | None | A filepath containing a plaintext version of an API token or other key to be used in an HTTP header to authenticate with the EEC. |
+| auth-header | Api-Token | A header that will be used to authenticate the provider with the EEC. |
+| refresh-interval | 10s | A time duration that defines how frequently the provider should check the given URL for updates. |
+| insecure | false | If set to "true", use HTTP for the connection to the server. If unset or set to "false", use HTTPS. |
```

Comment:
I think we should leave use out of public documentation if we're going to discourage direct use? We can either document internally or rely on the code comments in the config struct

Reply:
I'm fine with whatever, this just seemed like the ideal place to have config and the code comments may not be easy to locate. I think any sufficiently-motivated user could figure it out, so I was just going to rely on our disclaimer to discourage users. That said, I can strip down this file if we think it's a risk.